### DISCUSSION TITLE
HOW TO install apache-superset on docker (offline)

### DISCUSSION THREAD

**Discussion by rindy8707** [apache/superset#23904] [Q&A / Help]
My environment: without internet. centos.

I installed docker engine & docker-compose. and download apache-superset's sourcefile.
BUT "docker-compose -f docker-compose-non-dev.yml up" is NOT WORKING.
ERROR: response from daemon: get "https://registry-1.docker.io/v2" 

HELP ME!

**Comment by sebastianliebscher** [comment]
This does not work because the docker daemon tries to download the Superset, Redis and Postgres images from Dockerhub. 

You can start Superset components without an internet connection:
- git checkout to 2.1.0 release
- edit `docker-compose-non-dev.yml` and replace every occasion of `image: *superset-image` with `build: .`
- then run `docker-compose -f docker-compose-non-dev.yml build`

But then you still need Redis and Postgres. If you have those running, you can start Superset with
- `docker-compose -f docker-compose-non-dev.yml up`
